Why unused function will cause compile error ?

```shell
➜  wasmer git:(master) make build-capi-llvm-staticlib
-----------
INFORMATION
-----------

Host Target: `x86_64-apple-darwin`.
Enabled Compilers: llvm.
Testing the following compilers & engines:
  * API: llvm-universal llvm-dylib,
  * C-API: .
Cargo features:
  * Compilers: `--features llvm`.
Rust version: rustc 1.59.0 (9d1b2106e 2022-02-23).
NodeJS version: v16.14.0.
LLVM version: 13.0.1.


--------------
RULE EXECUTION
--------------


RUSTFLAGS="" cargo build --manifest-path lib/c-api/Cargo.toml --release \
		--no-default-features --features wat,staticlib,llvm,wasi,middlewares
   Compiling wasmer-c-api v2.2.1 (/Users/zhouyou/Downloads/2022/examples-wasmer/wasmer/lib/c-api)
error: function is never used: `get_default_compiler_config`
   --> lib/c-api/src/wasm_c_api/engine.rs:293:4
    |
293 | fn get_default_compiler_config() -> Box<dyn CompilerConfig> {
    |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> lib/c-api/src/lib.rs:20:5
    |
20  |     dead_code,
    |     ^^^^^^^^^

error: could not compile `wasmer-c-api` due to previous error
make: *** [build-capi-llvm-staticlib] Error 101
```

--- a/lib/c-api/src/wasm_c_api/engine.rs
+++ b/lib/c-api/src/wasm_c_api/engine.rs
@@ -289,7 +289,7 @@ pub struct wasm_engine_t {
 #[cfg(feature = "compiler")]
 use wasmer_api::CompilerConfig;
 
-#[cfg(feature = "compiler")]
+#[cfg(all(feature = "compiler", any(feature = "universal", feature = "dylib")))]
 fn get_default_compiler_config() -> Box<dyn CompilerConfig> {
     cfg_if! {
         if #[cfg(feature = "cranelift")] {

